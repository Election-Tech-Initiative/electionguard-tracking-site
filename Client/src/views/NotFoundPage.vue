<template>
  <div class="section">
    <div class="container">
      <div class="content is-size-3">
        <i18n path="notFoundMessage">
          <span class="is-inline-block has-text-weight-bold">{{ $t('notFound') }}</span>
        </i18n>
      </div>
      <div class="content">        
        <router-link class="button is-dark" to='/'>{{ $t('returnHome') }}</router-link>    
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

@Component
export default class NotFoundPage extends Vue {
}
</script>

<style>

</style>
