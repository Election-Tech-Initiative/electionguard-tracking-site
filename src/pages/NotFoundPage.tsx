import React from 'react';
import Title from '../components/Title';

const notFound = 'Not Found';

export interface NotFoundPageProps {}

const NotFoundPage: React.FunctionComponent<NotFoundPageProps> = () => <Title title={`404 ${notFound}`} />;

export default NotFoundPage;
