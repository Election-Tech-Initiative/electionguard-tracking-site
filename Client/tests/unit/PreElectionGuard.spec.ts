import { mount, RouterLinkStub } from '@vue/test-utils';

import NotFoundPage from '@/components/PreElectionGuard.vue';

describe('PreElectionGuard.vue', () => {
    it('renders preElectionMessage when election not closed', () => {
        const wrapper = mount(NotFoundPage, {
            mocks: {
                appSettings: {
                    electionClosed: 'false',
                    electionDate: 'June 20, 2019',
                },
                $t: (msg: string): string => msg
            },
            slots: {
                default: '<div />',
            },
        });
        expect(wrapper.element).toMatchSnapshot('election not closed');
    });

    it('renders slot div when election is closed', () => {
        const wrapper = mount(NotFoundPage, {
            mocks: {
                appSettings: {
                    electionClosed: 'true',
                    electionDate: 'June 20, 2019',
                },
                $t: (msg: string): string => msg
            },
            slots: {
                default: '<div />',
            },
        });
        expect(wrapper.element).toMatchSnapshot('election is closed');
    });
});
