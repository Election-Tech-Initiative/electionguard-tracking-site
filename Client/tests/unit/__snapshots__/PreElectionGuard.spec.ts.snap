// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreElectionGuard.vue renders preElectionMessage when election not closed: election not closed 1`] = `
<div>
  <div
    class="section"
  >
    <div
      class="container"
    >
      <div
        class="content"
      >
        <p>
          
              preElectionMessage
          
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`PreElectionGuard.vue renders slot div when election is closed: election is closed 1`] = `
<div>
  <div />
</div>
`;
